src/Microsoft.Framework.Runtime/Project.cs

```diff
@@ -324,86 +349,78 @@ private static SemanticVersion SpecifySnapshot(string version, string snapshotVa
                 {
-                    if (string.IsNullOrEmpty(dependency.Name))
+                    if (string.IsNullOrEmpty(dependencyKey))
                     {
-
                         throw FileFormatException.Create(
                             "Unable to resolve dependency ''.",
-                            dependency,
+                            dependencies.Value(dependencyKey),
                             projectPath);
                     }
 
-                    // Support 
-                    // "dependencies" : {
-                    //    "Name" : "1.0"
-                    // }
-
-                    var dependencyValue = dependency.Value;
+                    var dependencyValue = dependencies.Value(dependencyKey);
                     var dependencyTypeValue = LibraryDependencyType.Default;
+                    JsonString dependencyVersionAsString = null;
 
-                    string dependencyVersionValue = null;
-                    JToken dependencyVersionToken = dependencyValue;
-
-                    if (dependencyValue.Type == JTokenType.String)
+                    if (dependencyValue is JsonObject)
                     {
-                        dependencyVersionValue = dependencyValue.Value<string>();
-                    }
-                    else
-                    {
-                        if (dependencyValue.Type == JTokenType.Object)
-                        {
-                            dependencyVersionToken = dependencyValue["version"];
-                            if (dependencyVersionToken != null && dependencyVersionToken.Type == JTokenType.String)
-                            {
-                                dependencyVersionValue = dependencyVersionToken.Value<string>();
-                            }
-                        }
+                        // "dependencies" : { "Name" : { "version": "1.0", "type": "build" } }
+                        var dependencyValueAsObject = (JsonObject)dependencyValue;
+                        dependencyVersionAsString = dependencyValueAsObject.ValueAsString("version");
 
                         IEnumerable<string> strings;
-                        if (TryGetStringEnumerable(dependencyValue["type"], out strings))
+                        if (TryGetStringEnumerable(dependencyValueAsObject, "type", out strings))
                         {
                             dependencyTypeValue = LibraryDependencyType.Parse(strings);
                         }
                     }
+                    else if (dependencyValue is JsonString)
+                    {
+                        // "dependencies" : { "Name" : "1.0" }
+                        dependencyVersionAsString = (JsonString)dependencyValue;
+                    }
+                    else
+                    {
+                        throw FileFormatException.Create(
+                            "Unrecoganizable format of dependency version of " + dependencyKey,
```

Comment:
Ok just to be clear: I was joking about the spelling :smile: But the message overall could probably be improved as well (the phrasing is awkward and unclear).